static/src/javascripts/projects/common/modules/ui/subscription-banners/guardian-weekly-banner.spec.js

```diff
@@ -0,0 +1,280 @@
+                trackBannerView: jest.fn(),
+            };
+
+            const show = createBannerShow(
+                mockTrackingSUT,
+                gwBannerTemplate,
+                false
+            );
+            show();
+
+            expect(mockTrackingSUT.trackBannerView).toHaveBeenCalled();
+            expect(mockTrackingSUT.trackBannerView.mock.calls.length).toBe(1);
+        });
+
+        it('should call gaTracking for google analytics tracking', () => {
+            const mockTrackingSUT = { ...mockTracking, gaTracking: jest.fn() };
+
+            const show = createBannerShow(
+                mockTrackingSUT,
+                gwBannerTemplate,
+                false
+            );
+            show();
+
+            expect(mockTrackingSUT.gaTracking).toHaveBeenCalled();
+            expect(mockTrackingSUT.gaTracking.mock.calls.length).toBe(1);
+        });
+
+        it('should call trackBannerClick when cta ', () => {
+            const mockTrackingSUT = {
+                ...mockTracking,
+                trackBannerClick: jest.fn(),
+            };
+
+            const show = createBannerShow(
+                mockTrackingSUT,
+                gwBannerTemplate,
+                false
+            );
+            show();
+
+            const ctaButton = document.querySelector(bannerSelector('cta'));
+
+            if (ctaButton) {
+                ctaButton.click();
+            }
+
+            expect(mockTrackingSUT.trackBannerClick).toHaveBeenCalled();
+            expect(mockTrackingSUT.trackBannerClick.mock.calls.length).toBe(1);
+        });
+
+        it('should call trackBannerClick when sign-in', () => {
+            const mockTrackingSUT = {
+                ...mockTracking,
+                trackBannerClick: jest.fn(),
+            };
+
+            const show = createBannerShow(
+                mockTrackingSUT,
+                gwBannerTemplate,
+                false
+            );
+            show();
+
+            const signInLink = document.querySelector(
+                bannerSelector('sign-in')
+            );
+
+            if (signInLink) {
+                signInLink.click();
+            }
+
+            expect(mockTrackingSUT.trackBannerClick).toHaveBeenCalled();
+            expect(mockTrackingSUT.trackBannerClick.mock.calls.length).toBe(1);
+        });
+
+        it('should hide sign-in link if user is signed in', () => {
+            const mockTrackingSUT = {
+                ...mockTracking,
+                trackBannerClick: jest.fn(),
+            };
+            const isUserSignedIn = true;
+
+            const show = createBannerShow(
+                mockTrackingSUT,
+                gwBannerTemplate,
+                isUserSignedIn
+            );
+            show();
+
+            const signInContainer = document.querySelector(
+                '.site-message--subscription-banner__sign-in'
+            );
+
+            const hasSignedInClass: ?boolean =
```

Comment:
You don't need to keep the boolean type here anymore since the expression following it will always return a boolean. But if you want to keep it you should remove the optional `?` symbol since this expression will always definitely return a boolean value. Same thing goes for your other test file.

Reply:
this expression can now return null so I've kept the optional `?` symbol here

Reply:
The `&&` symbol will automatically change the expression's result into a boolean, despite any items in the expressions returning null.